src/lib/Form/Data/User/UserModeChangeData.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\Form\Data\User;
+
+final class UserModeChangeData
+{
+    private ?bool $mode;
```

Comment:
Why don't you use `UserMode` class constants for the value? It makes more sense since the setting is prepared to handle more user modes if there is ever the need. If your data class could reflect it, introducing more user modes will be easier as you would only need to change the form type.